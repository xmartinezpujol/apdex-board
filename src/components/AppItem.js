import Component from './Component';

class AppItem extends Component {
  constructor(props) {
    super();
    this.state = {
      name: props.name,
      version: props.version,
      apdex: props.apdex,
      contributors: props.contributors,
    }
    this.handleAlert = this.handleAlert.bind(this);
  }

  handleAlert(num) {
    alert(`Release number: ${num}`);
  }

  update(newState) {
    this.state.name = newState;
    document.getElementById(this._id).innerHTML = document.componentReg[this._id].render();
  }

  render() {
    return `
      <div id=${this._id}>
        <div
          onClick="document.componentReg[${this._id}].handleAlert(${this.state.version})"
          style="display: flex; margin: 23px 0px"
        >
          <div style="margin-right: 10px;">${this.state.apdex}</div>
          <div>${this.state.name}</div>
          <button
            style="margin-left: 20px;"
            onClick="document.componentReg[${this._id}].update('Lorem Ipsum')"
          >
            Click me to update
          </button>
        </div>
      </div>
    `;
  }
}

export default AppItem;
