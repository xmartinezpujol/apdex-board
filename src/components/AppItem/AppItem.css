.app-item {
    display: flex;
    padding: 15px 0;
}

.app-item:hover {
    background-color: red;
    cursor: pointer;
}

.app-item-apdex {
    margin-right: 10px;
}