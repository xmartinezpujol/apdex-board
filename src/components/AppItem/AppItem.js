import Component from './../Component';

import './AppItem.css';

class AppItem extends Component {
  constructor(props) {
    super();
    this.state = {
      name: props.name,
      version: props.version,
      apdex: props.apdex,
      contributors: props.contributors,
    };
    this.handleAlert = this.handleAlert.bind(this);
  }

  handleAlert() {
    alert(`Release number: ${this.state.version}`);
  }

  render() {
    return `
      <div
        class="app-item"
        onClick="document.componentReg[${this._id - document.currentState}].handleAlert()"
      >
        <div class="app-item-apdex">${this.state.apdex}</div>
        <div>${this.state.name}</div>
      </div>
    `;
  }
}

export default AppItem;
