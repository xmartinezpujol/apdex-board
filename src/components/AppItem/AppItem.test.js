import AppItem from './AppItem';
import mockAppItem from '../../data/MockDataApp';

const appItem = new AppItem(mockAppItem);
document.body.innerHTML = appItem.render();

describe('AppItem', () => {
  it('should handle the click event', () => {
    window.alert = jest.fn();
    document.querySelector('.app-item').onclick();
    expect(window.alert).toHaveBeenCalled();
  });

  it('has an apdex and is >= 0 & <= 100', () => {
    expect(appItem.state.apdex).toBeGreaterThan(0);
    expect(appItem.state.apdex).toBeLessThanOrEqual(100);
  });

  it('has a name', () => {
    expect(appItem.state.name).toBeDefined();
  });

  it('has a release version', () => {
    expect(appItem.state.version).toBeDefined();
  });

  it('renders correctly', () => {
    expect(appItem.render()).toMatchSnapshot();
  });
});
