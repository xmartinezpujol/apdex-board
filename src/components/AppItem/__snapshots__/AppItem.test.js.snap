// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppItem renders correctly 1`] = `
"
      <div
        class=\\"app-item\\"
        onClick=\\"document.componentReg[0].handleAlert()\\"
      >
        <div class=\\"app-item-apdex\\">100</div>
        <p>New Test App - Lorem - Ipsum, and Ipsums</p>
      </div>
    "
`;
